src/SAML2/SOAP.php

```diff
@@ -19,23 +17,24 @@ class SAML2_SOAP extends SAML2_Binding
      */
     public function send(SAML2_Message $message)
     {
-        assert('!empty($this->AssertionConsumerServiceURL)');
-        header('Content-Type: text/xml', TRUE);
-        $outputFromIdp = '<?xml version="1.0" encoding="UTF-8"?>';
-        $outputFromIdp .= '<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">';
-        $outputFromIdp .= '<SOAP-ENV:Header>';
-        $outputFromIdp .= '<ecp:Response xmlns:ecp="urn:oasis:names:tc:SAML:2.0:profiles:SSO:ecp" ';
-        $outputFromIdp .= '  SOAP-ENV:mustUnderstand="1" SOAP-ENV:actor="http://schemas.xmlsoap.org/soap/actor/next" ';
-        $outputFromIdp .= '  AssertionConsumerServiceURL="' . $this->AssertionConsumerServiceURL . '"/>';
-        $outputFromIdp .= '</SOAP-ENV:Header>';
-        $outputFromIdp .= '<SOAP-ENV:Body>';
-        $xmlMessage = $message->toSignedXML();
-        SAML2_Utils::getContainer()->debugMessage($xmlMessage, 'out');
-        $tempOutputFromIdp = $xmlMessage->ownerDocument->saveXML($xmlMessage);
-        $outputFromIdp .= $tempOutputFromIdp;
-        $outputFromIdp .= '</SOAP-ENV:Body>';
-        $outputFromIdp .= '</SOAP-ENV:Envelope>';
-        print($outputFromIdp);
+        
+        $envelope = '<soap-env:Envelope xmlns:soap-env="http://schemas.xmlsoap.org/soap/envelope/">'.
+                    '<soap-env:Header/><soap-env:Body /></soap-env:Envelope>';
+
+        $doc = new DOMDocument();
+        $doc->loadXML($envelope);
+        $soapHeader = $doc->getElementsByTagNameNS('http://schemas.xmlsoap.org/soap/envelope/', 'Header');
+        $soapBody = $doc->getElementsByTagNameNS('http://schemas.xmlsoap.org/soap/envelope/', 'Body');
+
+        if ($message->toSignedXML()->localName === 'Response') {
+            $response = new SAML2_XML_ecp_Response();
+            $response->AssertionConsumerServiceURL = $this->getDestination();
```

Comment:
`$this->destination` does not appear to be set. Not sure what the best way to set it off-hand is, but calling `$message->getDestination()` returns what we want.